packages/dd-trace/src/appsec/callbacks/ddwaf.js

```diff
@@ -79,7 +79,10 @@ class WAFCallback {
       wafContext = this.ddwaf.createContext()
     }
 
-    // TODO: if status code in params, convert it to string
+    // cast status code to string
+    if (params[addresses.HTTP_INCOMING_RESPONSE_CODE]) {
+      params[addresses.HTTP_INCOMING_RESPONSE_CODE] = params[addresses.HTTP_INCOMING_RESPONSE_CODE].toString()
```

Comment:
let's use string concatenation (`params[addresses.HTTP_INCOMING_RESPONSE_CODE] + ''`) as Number.prototype.toString might be impacted by side effects in the process. I checked, under the hood, concatenation does not call toString.

Reply:
```suggestion
      params[addresses.HTTP_INCOMING_RESPONSE_CODE] = params[addresses.HTTP_INCOMING_RESPONSE_CODE] + ''
```